site/app/models/RainbowCustomizationJSON.php

```diff
@@ -346,6 +361,31 @@ public function addPlagiarismEntry(object $plagiarismEntry): void {
     }
 
 
+    /**
+     * Add a manual grade entry to existing array
+     *
+     * @param object{
+     *     "user": string,
+     *     "grade": string,
+     *     "note": string
+     * } $manualGradeEntry
+     */
+    public function addManualGradeEntry(object $manualGradeEntry): void {
+        $emptyArray = [
+            "user" => "",
+            "grade" => "",
+            "note" => ""
+        ];
+
+        $manualGradeArray = json_decode(json_encode($manualGradeEntry), true);
```

Comment:
What's the goal here?  If you want to operate on an array, would it be better to accept an array instead?

Reply:
This is a somewhat convoluted way of casting to an array that I reused from the plagiarism table. Looking into it, it may work better than a cast for nested arrays. Changed to a cast in [f943457](https://github.com/Submitty/Submitty/pull/10653/commits/f9434572aabbd0b6632281e2ae47f55213025f7b)

Reply:
Can't you just make the function accept an array in the first place?

Reply:
The value the function is using needs to be an object before and after the function, so there would need to be a cast at some point (unless there's a better way to check whether every value in an object is empty?).